modules/django-drive/django_drive/drive/viewsets.py

```diff
@@ -0,0 +1,152 @@
+
+
+def init_service(token):
+    """
+    The method authorizes a user and creates an access point to interact with Google Drive.
+    Gives access to the user to access the files and folders of the drive
+    """
+    creds = Credentials(token=token)
+    return discovery.build('drive', 'v3', credentials=creds)
+
+
+class GetDriveFilesViewSet(APIView):
+    """
+    This class without any parameters return all the files and folders from the Google Drive. 
+    Filters results according to the provided queries.
+    """
+
+    def get(self, request, *args, **kwargs):
+        """
+        :header:
+            Authorization: send access token in header without keyword 'Bearer'
+        :query_param str query: To search for a specific set of files or folders, use the query. 
+        :param num pageSize: The maximum number of files to return per page
+        :param str pageToken: The token for continuing a previous list request on the next page. This should be set to the value of 'nextPageToken' from the previous response.
+        :param str mimeType: Specifies if the user wants to get files or folders supported values are `application/vnd.google-apps.file` and `application/vnd.google-apps.folder`
+        :return: Returns list of the files and folder from user's Google Drive.
+        """
+        try:
+            drive_service = init_service(token=request.META.get('HTTP_AUTHORIZATION'))
+            files = drive_service.files().list(
+                q=request.query_params.get('query'),
+                pageToken=request.data.get('pageToken'),
+                pageSize=request.data.get('pageSize')
+                ).execute()
+            return Response(files, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response(e.args, status=status.HTTP_400_BAD_REQUEST)
+
+
+class UploadDriveFileViewSet(APIView):
+    """
+    This class uploads file data on the Google Drive by making  multipart upload request.
+    """
+
+    def post(self, request, *args, **kwargs):
+        """
+        :header:
+            Authorization: send access token in header without keyword 'Bearer'
+        :param file: The file sent needs to be uploaded on the Google Drive. File must be sent as multipart/form-data
+        :return: Uploads the file to the specified folder and returns file id.
+        """
+        try:
+            drive_service = init_service(token=request.META.get('HTTP_AUTHORIZATION'))
+            serializer = UploadFileSerializer(data=request.FILES)
+            serializer.is_valid(raise_exception=True)
+
+            uploaded_file = request.FILES['file']
+            file_metadata = {
+                "name": uploaded_file.name,
+                "parents": [settings.PARENT_FOLDER]
+            }
+            media = apiclient.http.MediaInMemoryUpload(
+                body=uploaded_file.read(),
+                mimetype=uploaded_file.content_type)
+            file = drive_service.files().create(
```

Comment:
Also I think this Service structure is way to go for backend modules. That will held devs to use the service and reuse the menthods

in this case, these methods/features to create, share etc are part of some view that is not as modular as say the serevice here. SO, all the functional parts stay away from views and would only be called in views


https://github.com/crowdbotics/modules/pull/760/files#diff-085a3e324d641b786638b25b237f4bee23d50cb088d777d7c384dd7ae016d5af

Reply:
@shahraizali Sure, I'll move it service.